### ISSUE TITLE
react/jsx-indent-props 2 spaces with expression

### ISSUE THREAD

**Issue by noyobo** [yannickcr/eslint-plugin-react#1187]
- eslint v3.19.0
- eslint-plugin-react 6.10.3

![image](https://cloud.githubusercontent.com/assets/1292082/25927659/041f25d2-362a-11e7-8793-079aac8a0cba.png)

```
  111:15  error  Expected indentation of 12 space characters but found 14  react/jsx-indent-props
  112:15  error  Expected indentation of 12 space characters but found 14  react/jsx-indent-props
```

How to configure in the case?

**Comment by tnguyen14** [comment]
Running into this problem as well, after running `prettier` on the code base.

**Comment by tnguyen14** [comment]
The configuration I'm using is
```
"react/jsx-indent-props": [2, 2]
```

**Comment by TimonVS** [comment]
Are you extending from any other configs? If you're using Prettier I'd suggest using https://github.com/prettier/eslint-config-prettier to disable all stylistic rules (because Prettier already solves this for you).

**Comment by ljharb** [comment]
Prettier doesn't "solve" this; the goal is to be able to lint against it without modifying any files.

**Comment by noyobo** [comment]
I think so. `span` indentation is correct (110 lines In the screenshot). 

Why 111 and 112 indentation is error. They should relative to the previous line of code 2 spaces.

Otherwise:

```js
? <span
  className...
  dangerously...
  />
: null       
```

It looks very awkward

**Comment by ljharb** [comment]
I agree the props there should be indented 1 level in.

**Comment by yujiangshui** [comment]
Hi @ljharb, is this bug fixed? I see there are MRs but not be merged.

**Comment by ljharb** [comment]
#1317 isn't quite ready yet, according to its OP.

**Comment by davebream** [comment]
Anything new on this?

**Comment by ljharb** [comment]
The linked PR was closed, so this is again in need of someone willing to write a fix.

**Comment by lednhatkhanh** [comment]
Any fixes on this?

**Comment by yuritoledo** [comment]
+1

**Comment by golopot** [comment]
Maybe the current indent behavior is actually correct. I convinced myself when I compare it with other cases:

```javascript
// The following are all considered correct now.

const Foo1 = cond
  ? <div
    className={a}
    onClick={b}
  />
  : null;

const Foo2 = cond && (
  1 && <div
    className={a}
    onClick={b}
  />
);

const Foo3 = cond
  ? someFn(
    a,
    b
  )
  : null;
```

**Comment by yuritoledo** [comment]
The correct, (IMO) in this case:
```js 
const Foo1 = cond
  ? <div
    className={a}
    onClick={b}
  />
  : null;
```
Should be:
```js
const Foo1 = cond
  ? <div
        className={a}
        onClick={b}
     />
  : null;
 ```

**Comment by ljharb** [comment]
What I’d suggest to make it work best is to wrap multi line jsx in parens.

**Comment by Moong0122** [comment]
I will take a look this issue

**Comment by Hypnosphi** [comment]
@ljharb Now, if I use parens, jsx-indent-props wants me to use 4 spaces instead of the configured 2

```
    const child = selectedColor
      ? <h1 style={{color: selectedColor.key}}>{sayHello()}</h1>
      : (
        <EmptyWidget
          face={EmptyWidgetFaces.JOY}
          message={'Select "Edit..." option in widget dropdown to configure text color'}
        />
      );
```
>   94:11  error  Expected indentation of 12 space characters but found 10  react/jsx-indent-props
>  95:11  error  Expected indentation of 12 space characters but found 10  react/jsx-indent-props

**Comment by ljharb** [comment]
@Hypnosphi oops - a PR with that test case would help.
